### DISCUSSION TITLE
Homebrew not installed globally?

### DISCUSSION THREAD

**Discussion by jgarnie** [Homebrew/discussions#1203] [Getting started]
I just installed homebrew on linux mint 20 following the instructions here [https://brew.sh/](url)
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
`
also I installed the package:` brew install swagger-codegen`

it installed it on the folder 
`/home/linuxbrew/homebrew`

however when I go to a different folder and try to run swagger-codegen or brew help, it is telling me 
`bash: brew: command not found
`
I dont understand what I am missing since according to installation it should be all, and using `brew help` on the installation folder actually shows results.

Does anyone had similar issue with brew?

thanks for your time

**Comment by carlocab** [comment]
Did you add `brew` to your `PATH`, as suggested by the installer when it finished?

**Reply by jgarnie** [reply]
thanks for your reply and help.

OK, this can be a problem I believe, the way I did it just copied and pasted it on the terminal once again, is that correct?
1
` echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' >> /home/joaquin/.profile
`   
2
` eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
`
3
`sudo apt-get install build-essential
`
4` brew install gcc
`

following the instructions on the consoole:

`=> Next steps:
- Add Homebrew to your PATH in /home/joaquin/.profile:
    echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' >> /home/joaquin/.profile
    eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
- Run `brew help` to get started
- Further documentation: 
    https://docs.brew.sh
- Install the Homebrew dependencies if you have sudo access:
    sudo apt-get install build-essential
    See https://docs.brew.sh/linux for more information
- We recommend that you install GCC:
    brew install gcc
`

is it correct or am I missing sth?

**Reply by carlocab** [reply]
That looks right. You might have duplicate lines in your `~/.profile` that you may wish to clean up though. (Keeping them shouldn’t hurt.)

What’s the output of `echo $SHELL`?

Try restarting your terminal.

**Reply by jgarnie** [reply]
after running:
echo $SHELL
the output ` /bin/bash`

still same result when I try brew help outside the installation folder and after restart:
bash: brew: command not found

any clue what is wrong here?

**Reply by carlocab** [reply]
What's the output of `echo $PATH`?

**Reply by jgarnie** [reply]
output : 
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

**Reply by carlocab** [reply]
Ok, your problem is that `/home/linuxbrew/.linuxbrew/bin` isn't in your `PATH`. Not sure why, as that should be put there by having 
```
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
```
in your `~/.profile`. Can you do
```
cat ~/.profile
```
and check whether the `eval` line above shows up there? If it appears there, check if any of the following files exist in `$HOME`: `.bash_profile`, `.bash_login`, or `.bashrc`.

**Reply by jgarnie** [reply]
After checking cat ~/.profile, I see the following
```
 ~/.profile: executed by the command interpreter for login shells.
# This file is not read by bash(1), if ~/.bash_profile or ~/.bash_login
# exists.
# see /usr/share/doc/bash/examples/startup-files for examples.
# the files are located in the bash-doc package.

# the default umask is set in /etc/profile; for setting the umask
# for ssh logins, install and configure the libpam-umask package.
#umask 022

# if running bash
if [ -n "$BASH_VERSION" ]; then
    # include .bashrc if it exists
    if [ -f "$HOME/.bashrc" ]; then
	. "$HOME/.bashrc"
    fi
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" ] ; then
    PATH="$HOME/bin:$PATH"
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"

```
it appears 5 times, I blieve is because I tried several times

in my Home I see the file .bashrc

**Reply by jgarnie** [reply]
I see that when I am in the installation folder 'home' and run  echo $PATH, I get

`/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin`

however as soon as I open a new terminal this changes to 
echo $PATH
`/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin`
even in my home folder where in the other terminal see the correct path

**Reply by carlocab** [reply]
Sounds like you have something in your terminal that changes your `PATH` when you navigate to certain directories. That's definitely non-standard, and there are many different reasons it could be happening (e.g. you have something like `direnv` with an `.envrc`).

You'll need to work out what's changing your `PATH` when you navigate away from your home directory and fix that.

I'd look in `.bashrc` to see if there's anything fishy there.

**Reply by jgarnie** [reply]
thank you very much for your help, I will dfinetly have a look to it, it is inded a good analysis and support from your side.

just in case you know what is happening here is my .bashrc file.
 cat .bashrc
```
# ~/.bashrc: executed by bash(1) for non-login shells.
# see /usr/share/doc/bash/examples/startup-files (in the package bash-doc)
# for examples

# If not running interactively, don't do anything
case $- in
    *i*) ;;
      *) return;;
esac

# don't put duplicate lines or lines starting with space in the history.
# See bash(1) for more options
HISTCONTROL=ignoreboth

# append to the history file, don't overwrite it
shopt -s histappend

# for setting history length see HISTSIZE and HISTFILESIZE in bash(1)
HISTSIZE=1000
HISTFILESIZE=2000

# check the window size after each command and, if necessary,
# update the values of LINES and COLUMNS.
shopt -s checkwinsize

# If set, the pattern "**" used in a pathname expansion context will
# match all files and zero or more directories and subdirectories.
#shopt -s globstar

# make less more friendly for non-text input files, see lesspipe(1)
[ -x /usr/bin/lesspipe ] && eval "$(SHELL=/bin/sh lesspipe)"

# set variable identifying the chroot you work in (used in the prompt below)
if [ -z "${debian_chroot:-}" ] && [ -r /etc/debian_chroot ]; then
    debian_chroot=$(cat /etc/debian_chroot)
fi

# set a fancy prompt (non-color, unless we know we "want" color)
case "$TERM" in
    xterm-color|*-256color) color_prompt=yes;;
esac

# uncomment for a colored prompt, if the terminal has the capability; turned
# off by default to not distract the user: the focus in a terminal window
# should be on the output of commands, not on the prompt
#force_color_prompt=yes

if [ -n "$force_color_prompt" ]; then
    if [ -x /usr/bin/tput ] && tput setaf 1 >&/dev/null; then
	# We have color support; assume it's compliant with Ecma-48
	# (ISO/IEC-6429). (Lack of such support is extremely rare, and such
	# a case would tend to support setf rather than setaf.)
	color_prompt=yes
    else
	color_prompt=
    fi
fi

if [ "$color_prompt" = yes ]; then
    PS1='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]\$ '
else
    PS1='${debian_chroot:+($debian_chroot)}\u@\h:\w\$ '
fi
unset color_prompt force_color_prompt

# If this is an xterm set the title to user@host:dir
case "$TERM" in
xterm*|rxvt*)
    PS1="\[\e]0;${debian_chroot:+($debian_chroot)}\u@\h: \w\a\]$PS1"
    ;;
*)
    ;;
esac

# enable color support of ls and also add handy aliases
if [ -x /usr/bin/dircolors ]; then
    test -r ~/.dircolors && eval "$(dircolors -b ~/.dircolors)" || eval "$(dircolors -b)"
    alias ls='ls --color=auto'
    #alias dir='dir --color=auto'
    #alias vdir='vdir --color=auto'

    alias grep='grep --color=auto'
    alias fgrep='fgrep --color=auto'
    alias egrep='egrep --color=auto'
fi

# colored GCC warnings and errors
#export GCC_COLORS='error=01;31:warning=01;35:note=01;36:caret=01;32:locus=01:quote=01'

# some more ls aliases
alias ll='ls -alF'
alias la='ls -A'
alias l='ls -CF'

# Add an "alert" alias for long running commands.  Use like so:
#   sleep 10; alert
alias alert='notify-send --urgency=low -i "$([ $? = 0 ] && echo terminal || echo error)" "$(history|tail -n1|sed -e '\''s/^\s*[0-9]\+\s*//;s/[;&|]\s*alert$//'\'')"'

# Alias definitions.
# You may want to put all your additions into a separate file like
# ~/.bash_aliases, instead of adding them here directly.
# See /usr/share/doc/bash-doc/examples in the bash-doc package.

if [ -f ~/.bash_aliases ]; then
    . ~/.bash_aliases
fi

# enable programmable completion features (you don't need to enable
# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
# sources /etc/bash.bashrc).
if ! shopt -oq posix; then
  if [ -f /usr/share/bash-completion/bash_completion ]; then
    . /usr/share/bash-completion/bash_completion
  elif [ -f /etc/bash_completion ]; then
    . /etc/bash_completion
  fi
fi

```

in case you might know if there is sth that shouldnt be here.
Thank you very much

**Reply by carlocab** [reply]
Nope, that all looks fine. Not sure I can be much more help from here. Sorry!

**Reply by gromgit** [reply]
Just to be sure: What shell are you running? bash? zsh? something else?

**Reply by carlocab** [reply]
`echo $SHELL` returns `/bin/bash`. https://github.com/Homebrew/discussions/discussions/1203#discussioncomment-574166

**Reply by jgarnie** [reply]
exactly, /bin/bash is what I get after $SHELL

**Reply by gromgit** [reply]
Sorry, missed that. Anyway, it's probably not critical to the issue.

@jgarnie, I assume you have _not_ logged out and back in again since you installed Homebrew? If so, what you've described so far is in accordance with the fact that your `brew shellenv` lines were added to `~/.profile`, which is only sourced by a login shell. Opening a new terminal window merely starts an interactive non-login shell, which sources `~/.bashrc` instead.

Aside from cleaning up the multiple `brew shellenv` lines in your `~/.profile`, I think a simple logout and login should sort things out.

**Reply by carlocab** [reply]
That doesn't explain why `$HOMEBREW_PREFIX/bin` is in their `PATH` when they're in their `HOME` directory but not otherwise, though. Still, could be worth a try.

**Reply by gromgit** [reply]
The way I read https://github.com/Homebrew/discussions/discussions/1203#discussioncomment-574445, the first part was done in the installation window (where `eval "$(brew shellenv)"` was previously run manually), while the second part was in a new window.

**Reply by carlocab** [reply]
Aha, yes. I misread that. I agree with your assessment then.

**Reply by jgarnie** [reply]
Hi everyone, apparenty this morning after switching on again the PC, everything was working as expected, so perhaps just a restart was needed?.

thanks to you bothh for the hard work and the help provided

**Reply by carlocab** [reply]
Yes, that's essentially what @gromgit was suggesting you do above. Glad it works now.
